index.js

```diff
@@ -32,7 +37,11 @@ module.exports = (input, opts) => new Promise((resolve, reject) => {
 				} else if (err instanceof TypeError) {
 					operation.stop();
 					reject(err);
-				} else if (!operation.retry(err)) {
+				} else if (operation.retry(err)) {
+					err.attemptNo = attemptNo;
```

Comment:
It should be named `attemptNumber`, not `attemptNo`. Applies to everywhere `attemptNo` is currently used.